src/test/java/organice/logic/commands/MatchCommandTest.java

```diff
@@ -0,0 +1,351 @@
+        MatchCommand matchCommand = new MatchCommand(VALID_NRIC_PATIENT_IRENE);
+
+        boolean matchResult = matchCommand.match(DONOR_IRENE_DONOR, PATIENT_IRENE);
+        assertTrue(matchResult); //100% match and same blood type
+
+        //donor with 4/6 matching tissues and same blood type.
+        matchResult = matchCommand.match(DONOR_SIMILAR_TISSUE, PATIENT_IRENE);
+        assertTrue(matchResult);
+
+        //donor with blood type A matches patient with blood type A and AB
+        assertEquals(matchCommand.match(DONOR_A, PATIENT_A), true);
+        assertEquals(matchCommand.match(DONOR_A, PATIENT_AB), true);
+
+        //donor with blood type B matches patient with blood type B and AB
+        assertEquals(matchCommand.match(DONOR_B, PATIENT_B), true);
+        assertEquals(matchCommand.match(DONOR_B, PATIENT_AB), true);
+
+        //donor with blood type AB matches patient with blood type AB
+        assertEquals(matchCommand.match(DONOR_AB, PATIENT_AB), true);
+
+        //donor with blood type O matches patient with blood type A, B, AB and O
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_A), true);
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_B), true);
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_AB), true);
+        assertEquals(matchCommand.match(DONOR_O, PATIENT_O), true);
+    }
+
+    @Test
+    public void match_isNotSuccessfulMatch() throws Exception {
+        MatchCommand matchCommand = new MatchCommand(VALID_NRIC_PATIENT_IRENE);
+
+        Donor differentBloodType = new DonorBuilder(DONOR_IRENE_DONOR).withBloodType("B").build();
+        boolean matchResult = matchCommand.match(differentBloodType, PATIENT_IRENE);
+        assertFalse(matchResult); //100% tissue match and different blood type
+
+        //donor with 4/6 matching tissues and different blood type.
+        //Hardcoded because there is no way to change tissue types in a more elegant manner
+        Donor someMatchingTissues = new DonorBuilder(DONOR_IRENE_DONOR).withTissueType(COMPATIBLE_TISSUE_TYPE_IRENE)
+                .withBloodType("B").build();
+        matchResult = matchCommand.match(someMatchingTissues, PATIENT_IRENE);
+        assertFalse(matchResult);
+
+        //donor with blood type A do not match patients with blood type B and O
+        assertEquals(matchCommand.match(DONOR_A, PATIENT_B), false);
+        assertEquals(matchCommand.match(DONOR_A, PATIENT_O), false);
+
+        //donor with blood type B do not match patients with blood type A and P
+        assertEquals(matchCommand.match(DONOR_B, PATIENT_A), false);
+        assertEquals(matchCommand.match(DONOR_B, PATIENT_O), false);
+
+        //donor with blood type AB do not match patients with blood type A, B and O
+        assertEquals(matchCommand.match(DONOR_AB, PATIENT_A), false);
+        assertEquals(matchCommand.match(DONOR_AB, PATIENT_B), false);
+        assertEquals(matchCommand.match(DONOR_AB, PATIENT_O), false);
+    }
```

Comment:
Same as above but opposite, should we test if there is a match in bloodtype

Reply:
This method is for testing for unsuccessful matches. Testing for successful matches are in the `match_isSuccessfulMatch()` method